### DISCUSSION TITLE
Questions and Feedback concerning 1.75.

### DISCUSSION THREAD

**Discussion by gaveitatry2** [r57zone/DualShock4-emulator#42] [General]
1.) This should be improved so that Alt + Tabbing to another window causes the cursor to be displayed again.  I switch windows a lot and I like being able to see my cursor when I am in other windows.  I don't want to exit out of DS4Emulator every single time that I want my cursor back.  But that's okay.  I already had a good program that lets me toggle cursor visibility on and off with a hotkey.  I will just use that.  I know that a lot of people say things like "don't worry" and "it's okay" but secretly they care a lot and they are just hiding their true feelings.  That is not the case for me.  I am happy to use the toggle cursor visibility program that I have.  It's an AutoHotKey script that I found a long time ago and it works wonderfully (I like it better than your old HideCursorTool program).  And I noticed that you told someone else that "there is a problem that won't let you hide/display the cursor multiple times".  So I understand.

2.) I noticed that you have both ActivateOnlyInWindow and ActivateOnlyInWindow2 now.  Thank you for adding ActivateOnlyInWindow2.  It's reassuring to know that I can make DS4Emulator work with both "PlayStation™Now" and "PlayStation Now".

3.) A long time ago, you told me that you made some adjustments to the touchpad presses and swipes.  I tested it out just now in Until Dawn.  It's still not very responsive.  And I still prefer to use the touchpad presses instead of the touchpad swipes, because it works better and takes less tries to work.  It feels like I have to keep spamming the touchpad swipe keys many times to get it to work.  But I don't really care about this.  I'm just happy that you implemented this feature and that it works.  It saved me when I was playing Until Dawn earlier this year.  It gets the job done.  If there is a game where you have a two second time limit to do a swipe, then I might be in trouble.  But I haven't encountered any games like that.  Until Dawn was the only game I had played that required touchpad swipes, and it was for unimportant things (no rush) like turning a lighter on or looking at game decisions history.

4.) This is a big bug.  Whenever I press the PlayStation Button key to access the menu (Return to Game, Close Game, Application Saved Data Management, Vibration Settings) it usually always brings my Steam window to the front.  I've been testing this out for hours, and the problem won't go away.  If I press the PlayStation Button, I want the PlayStation menu to show up but I don't want any other computer windows like Steam going to the front and covering up my PlayStation Now window.  I did try to minimize Steam to see if that helped, but it would still bring the Steam window up to the front.  And I don't want to completely exit the Steam program (haven't tried it, but don't want to do it).

5.) What does TOUCHPAD_SECOND_RIGHT do?  How is it different from TOUCHPAD_RIGHT?  I noticed that TOUCHPAD_SECOND_RIGHT can't be used for swipes.  I have

H for "Touchpad Left".
K for "Touchpad Right."
I for "Touchpad Second Right".

If I press K + H, I can swipe left (from right to left).  But if I press I + H, nothing happens.  So I think that TOUCHPAD_SECOND_RIGHT is for something else, but I don't know what it is used for.

6.) What are some examples of when I will need to use the LS (Left Stick)?  In DS4Emulator 1.6, I was able to play games using only the O/X/△/□, L1/L2/L3/R1/R2/R3 keys, and DPAD.  I never used the LS before because I never owned a PS or Xbox controller, and DS4Emulator 1.6 didn't have keys for LS.  I just want an idea of what I need the LS for because I am running out of buttons on my keyboard to use for the PlayStation controller.

7.) If there are controls that I don't need or don't want to use, what are some ways that I can disable them so they won't use up all of my keyboard keys?

a.) Assign them all to the same key that I don't use?
```
LS_UP=222
LS_DOWN=222
LS_LEFT=222
LS_RIGHT=222
```

b.) Is it possible to just delete the those lines for the keys?

c.) Is it possible to comment out the LS lines, or any other keys that I don't use?
```
# LS_UP=87
# LS_DOWN=83
# LS_LEFT=65
# LS_RIGHT=68
```

8.) I am not interested in Xbox Controllers, so I was wondering if I could completely delete this code?  Or do I need to leave it?

```
[Xbox]
SwapTriggersShoulders=0
SwapShareTouchPad=0
SleepTimeOut=1
TouchPadPressedWhenSwiping=0
# ------ Alt + F10 to get drift values.
DeadZoneLeftStickX=0
DeadZoneLeftStickY=0
DeadZoneRightStickX=0
DeadZoneRightStickY=0
```

9.) This is a question.  When I press the StopCenteringKey, I need to hold down the button, right?  I have been holding it down instead of pressing once, and I think that is good.  But I just wanted to make sure that this is how you intended it to be.  I reworded the comment above the code like this so I won't forget to (hold down) the key.

```
# ------ F2 (hold down) to stop centering.
StopCenteringKey=113
```

10.) What is the Button Code for "\"?  I looked on https://github.com/r57zone/Half-Life-Alyx-novr/blob/master/BINDINGS.md and that is the only key that doesn't have a button code for it.

11.) Is FullScreenTopOffset only for people who use FullScreen?  (I don't play PlayStation Now games in FullScreen).  And what does this affect exactly?  Does it affect mouse visibility?  Or does it affect the mouse's "center of PlayStation window".  Or something else?  I don't need it, but I just want to understand what it does exactly.

12.) I haven't tested Motion yet.  I will play The Last of Us Part II today and try to recharge the flashlight.  I will let you know how it works.  Thank you so much for adding this feature.

**Comment by r57zone** [comment]
1. There is a problem that does not allow hiding to show the cursor more than 1 time after the start of the application. I think I'll make hiding the cursor also by hotkey, so that it's more intuitive when you first start.
3. While I don't know what the problems might be, I'm writing blindly, without testing, so so far, but it seems I've tried everything.
4. Some users asked for this function, maybe to close the steam?
5. The "touchpad second right" was the second finger, for the Infamous: Second Son test, but this functionality has already been removed in the latest version, because it seems not to work.
6. LS is the Left Stick, by default it is WASD, but someone seems to have asked for a change to these buttons and I added this. No additional buttons are required.
7. You can assign unnecessary elements to one unnecessary, unused key. 
a) The Left Stick is used to move the player and it is needed.
b) The lines can be deleted, but then the default values will be used, the same buttons that I had in the config.
c) You can delete, but not all setiings, button are possible to delete.
8. You can, if it concerns the config.
9. Yep.
10. There is "/ or ? - 191", each button has a unique value.
11. This affects the window shift to the top, where the title of the PlayStation window is written. This is necessary so that he does not distract from the game title.
Thanks for the feedback.

**Comment by gaveitatry2** [comment]
2.) This is not a bug.  I was just thanking you for letting us add more than one ActivateOnlyInWindow.  I asked you over six months ago to have more than one, because I wanted both "PlayStation™Now" and "PlayStation Now".  Thank you for adding this.

4.) Making the Steam window come to the front every time that someone presses the PlayStation Button is stupid.  Can you at least make it optional?  I don't want this feature.

5.) You said that you removed the "touchpad second right".  Do you mean in the unreleased 1.7.6?  1.7.5 still has "touchpad second right".

10.) I am not asking about the "`/`" or "`?`" button code.  I am asking about the "`\`" key.  https://github.com/r57zone/Half-Life-Alyx-novr/blob/master/BINDINGS.md  There isn't a button code for it.

11.) I'm not sure I understand.  But that's okay.  I don't use Fullscreen anyway.  I already hide the title bar in every game that I play in windowed mode, including PlayStation Now.  But I don't understand what this has to do with Fullscreen mode.


Well, I purchased my PlayStation Now 1-month subscription yesterday, and I'll start gaming later today.  I can't wait to play and use your DS4Emulator.  Have a good day!

**Reply by r57zone** [reply]
4 - You can simply close the steam during the game.
5 - Already removed in 1.7.5
10 - Fixed
11 - ![](https://user-images.githubusercontent.com/9499881/144131491-7770060d-71be-492a-9c78-047691b16bc8.jpg)
Have a good day!

**Reply by gaveitatry2** [reply]
4 - I had to completely exit out of the Steam program to make the window stop coming to the front.  But, I don't like doing this for several reasons.  One, I like having Steam open so I can receive and respond to Steam IM's from friends.  Two, I have over 300 games on Steam, and the icons do not load for all of the games unless I take the time to slowly scroll down the long list and give the icons time to load.  I use a macro to do this.  But because of DS4Emulator, I have to exit Steam while playing PlayStation games.  Then restart Steam once I am done playing PlayStation games.  Then use the macro to restart Steam a second time so I can rename some games and auto scroll down the list to make the icons load.  A lot of work.  Because 0.0001% people want Steam to come to the front when you press the PlayStation Now button?  Sometimes I press that button by accident and it is annoying to make Steam come to the front each time.  I know you can make this optional or use a BringWindowToFront=(window name).  In fact that might be better.  What if someone uses Epic Games instead of Steam?  Or they prefer bringing their browser window to the front?  If you make it an option it will make more people happy and less people sad.  Everyone wins, except it requires more work.  But you will end up with a better program as a result.

5.) This is the original config file for 1.75 (Portable).  "TOUCHPAD_SECOND_RIGHT" has not be removed yet.

```
# Touchpad
TOUCHPAD_UP=85
TOUCHPAD_DOWN=78
TOUCHPAD_LEFT=72
TOUCHPAD_RIGHT=75
TOUCHPAD_CENTER=74
TOUCHPAD_SECOND_RIGHT=73
```

10.) Thanks for adding the button code!  I thought that list belonged to someone else.  I had no idea that it was yours.

12.) I wasn't able to test "Motion" out yet because The Last of Us Part II doesn't require you to recharge your flashlight.  I will have to test it out in The Last of Us (first game) later.  Not sure when.

**Reply by gaveitatry2** [reply]
4.) - I just found out that I can use the Esc button on my keyboard to open the PlayStation menu (Return to Game, Close Game, Application Saved Data Management, Vibration Settings) instead of using the PlayStation Button.  The Esc button doesn't bring the Steam window to the front, so that is good.  I just won't press the PlayStation Button anymore.

**Comment by gaveitatry2** [comment]
One way you can handle the Steam window coming to the front is have something like BringWindowToFront=Steam.  That way people can change which window they want or they can make it so that no window comes to the front.

**Comment by gaveitatry2** [comment]
>12.) I haven't tested Motion yet. I will play The Last of Us Part II today and try to recharge the flashlight. I will let you know how it works. Thank you so much for adding this feature.

Oh, I wanted to thank you for the Motion control.  I don't know how it works.  And I haven't tested it in the first TLOU game.  But in TLOU2, in Chapter 2 or 3, I needed to recharge my flashlight and it worked!!!  Thank you so much!

**Comment by r57zone** [comment]
I added disabling the Xbox button, in the config file need to change "EnableXboxButton = 1" to "EnableXboxButton = 0" and it will stop working.

**Reply by gaveitatry2** [reply]
Hey r57zone.  I'm a weirdo going through some changes.  I deleted all of my PC games and quit playing The Last of Us Part II around December 19, 2021.  I realized that I wasted my life doing nothing but sitting on my computer all day playing games and watching videos.  I guess a tiny part of me is sad that I had so many games that I did not finish, games that I had not played yet, games that will come out in the future.  And the same goes for anime and TV shows, etc.  But I feel like it wasn't adding to my life.  It was taking away.  So I am saying goodbye to it all.  Thank you for your great software.  I know a lot of people will benefit from having an emulator that will let them use their keyboard and mouse instead of an actual controller.  Take care.
